app/models/attachment_reference.rb

```diff
@@ -0,0 +1,61 @@
+class AttachmentReference < ActiveRecord::Base
+  after_initialize :set_expiry_at, if: proc { new_record? && !attachable }
```

Comment:
do not define callbacks using procs and make it do anything except call a method. extract this as a method.